presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -110,6 +110,7 @@
 import static io.prestosql.plugin.hive.util.HiveUtil.getHeaderCount;
```

Comment:
what does this commit solve?

Reply:
The current implementation does a listing per file, which causes excess latency. This reduces the number of listings needed, especially if files are all contained within the same directory (which is the common pattern).